src/mlpack/tests/main_tests/kmeans_test.cpp

```diff
@@ -0,0 +1,414 @@
+  double P = 2.0;
+  arma::mat inputData;
+  if (!data::Load("vc2.csv", inputData))
+    BOOST_FAIL("Unable to load train dataset vc2.csv!");
+
+  SetInputParam("input", std::move(inputData));
+  SetInputParam("refined_start", true);
+  SetInputParam("clusters", c);
+  SetInputParam("percentage", std::move(P));     // Invalid
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+
+/**
+ * Checking percentage is non-negative when --refined_start is specified
+ */
+BOOST_AUTO_TEST_CASE(NonNegativePercentageTest)
+{
+  int c = 2;
+  double P = -1.0;
+  arma::mat inputData;
+  if (!data::Load("vc2.csv", inputData))
+    BOOST_FAIL("Unable to load train dataset vc2.csv!");
+
+  SetInputParam("input", std::move(inputData));
+  SetInputParam("refined_start", true);
+  SetInputParam("clusters", c);
+  SetInputParam("percentage", P);     // Invalid
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+
+/**
+ * Checking that size and dimensionality of prediction is correct.
+ */
+BOOST_AUTO_TEST_CASE(KmClusteringSizeCheck)
+{
+  int c = 2;
+  arma::mat inputData;
+  if (!data::Load("vc2.csv", inputData))
+    BOOST_FAIL("Unable to load train dataset vc2.csv!");
+
+  size_t col = inputData.n_cols;
+  size_t row = inputData.n_rows;
+
+  SetInputParam("input", std::move(inputData));
+  SetInputParam("clusters", c);
+  SetInputParam("allow_empty_clusters", false);
+
+  mlpackMain();
+
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("output").n_rows, row+1);
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("output").n_cols, col);
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("centroid").n_rows, row);
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("centroid").n_cols, c);
+}
+
+/**
+ * Checking that size and dimensionality of prediction is correct when --labels_only is specified
+ */
+BOOST_AUTO_TEST_CASE(KmClusteringSizeCheckLabelOnly)
+{
+  int c = 2;
+
+  arma::mat inputData;
+  if (!data::Load("vc2.csv", inputData))
+    BOOST_FAIL("Unable to load train dataset vc2.csv!");
+  size_t col = inputData.n_cols;
+  size_t row = inputData.n_rows;
+
+  SetInputParam("input", std::move(inputData));
+  SetInputParam("clusters", c);
+  SetInputParam("labels_only", true);
+  SetInputParam("allow_empty_clusters", false);
+
+  mlpackMain();
+
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("output").n_rows, 1);
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("output").n_cols, col);
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("centroid").n_rows, row);
+  BOOST_REQUIRE_EQUAL(CLI::GetParam<arma::mat>("centroid").n_cols, c);
+}
+
+
+/**
+ * Checking that predictions are not same when --allow_empty_clusters or kill_empty_clusters are specified
+ */
+BOOST_AUTO_TEST_CASE(KmClusteringEmptyClustersCheck)
+{
+  int c = 400;
+  int iterations = 100;
+
+  arma::mat inputData;
+  if (!data::Load("vc2.csv", inputData))
```

Comment:
Try this one with `test_data_3_1000.csv`.  I think maybe the VC2 dataset has too much structure to produce significantly different results.

Reply:
Yup you are right VC2 dataset was not suitable for the test.